### PR TITLE
Refactor unauthorized front-end

### PR DESCRIPTION
**Pull request by nielsenramon**
Big refactor and redesign of the not-logged-in pages to improve usability and aesthetics.

## TODO

- [x] Upgrade to Bootstrap V4
- [x] Properly overwrite Bootstrap
- [x] New manifest file called unauthorized
- [x] Better UX and clarity in authorization flow (delcarative titles, readable forms, ...)
- [x] Responsive design
- [x] Add SCSS linting and csscomb config
- [x] Reduce variable complexity
- [x] Redesign register page
- [x] Redesign login page
- [x] Redesign password reset
- [x] Redesign notifications

## Impressions

screen shot 2015-11-12 at 21 58 06
screen shot 2015-11-12 at 21 58 19
screen shot 2015-11-12 at 21 58 33
screen shot 2015-11-12 at 21 55 09

### CODE REVIEW

File: app/views/locomotive/passwords/new.html.slim [hunk 1]
```diff
@@ -1,17 +1,20 @@
-- title current_site ? current_site.name : t('locomotive.not_logged_in.title')
+- title t('locomotive.passwords.new.title')
 
-= locomotive_form_for(resource, as: resource_name, url: locomotive_account_password_path) do |f|
+.public-form
+  .public-form-text
+    h3= t('locomotive.passwords.new.title')
+    p= t('locomotive.passwords.new.description')
 
-  = f.hidden_field :reset_password_token
```
Comment (did) [inline app/views/locomotive/passwords/new.html.slim]: we don't need it anymore?

File: app/assets/javascripts/locomotive/not_logged_in.js.coffee [hunk 2]
```diff
@@ -9,3 +9,20 @@ window.Locomotive = {}
 $ ->
   _.each window.flash_messages, (couple) ->
       Locomotive.notify couple[1], couple[0]
+
+  # JavaScript to render the form wrapper vertically in the middle.
```
Comment (did) [inline app/assets/javascripts/locomotive/not_logged_in.js.coffee]: why not using flex properties?

File: app/views/locomotive/layouts/not_logged_in.html.slim [hunk 3]
```diff
@@ -1,14 +1,19 @@
...
+
     meta name="viewport" content="width=device-width, initial-scale=1"
 
     link rel="icon" type="image/png" href="#{asset_url('locomotive/favicon.png')}" sizes="32x32"
 
+    link href="https://fonts.googleapis.com/css?family=Muli:400,300|Noto+Sans:400,700" rel="stylesheet" type="text/css"
+
     = javascript_include_tag  'locomotive/not_logged_in'
-    = stylesheet_link_tag     'locomotive/not_logged_in', media: 'screen'
+    = stylesheet_link_tag     'locomotive/unauthorized', media: 'screen'
```
Comment (did) [inline app/views/locomotive/layouts/not_logged_in.html.slim]: better name :-)

File: config/locales/nl.yml [hunk 4]
```diff
@@ -1,7 +1,5 @@
 nl:
   locomotive:
-    not_logged_in:
-      title: "Locomotive <small>Designed to save your time</small>"
```
Comment (did) [inline config/locales/nl.yml]: The localization workflow is to first add keys in english and then use the Transifex command line too to push those new keys to their platform. I assume all the translators get notified in order to fill them.
Finally, I use another Transifex command to pull the new versions in the project.

File: app/assets/javascripts/locomotive/not_logged_in.js.coffee (hunk 2 above)
Comment (nielsenramon) [inline app/assets/javascripts/locomotive/not_logged_in.js.coffee]: I still think flex is not supported widely enough. IE 8, 9 and 10 is not supported. neither are older android browsers: http://caniuse.com/#search=flex (also older safari versions do not support it).

It's your call :)

File: app/views/locomotive/passwords/new.html.slim (hunk 1 above)
Comment (nielsenramon) [inline app/views/locomotive/passwords/new.html.slim]: It's still in there :) see line 10.

**Pull request merged**